azurerm/internal/services/network/express_route_circuit_connection_resource.go

```diff
@@ -186,8 +186,10 @@ func resourceExpressRouteCircuitConnectionRead(d *pluginsdk.ResourceData, meta i
 	if props := resp.ExpressRouteCircuitConnectionPropertiesFormat; props != nil {
 		d.Set("address_prefix_ipv4", props.AddressPrefix)
 
+		// The ExpressRoute Circuit Connection API returns "*****************" for AuthorizationKey when it's changed from a valid value to `nil`
+		// See more details from https://github.com/Azure/azure-rest-api-specs/issues/15030
 		authorizationKey := ""
-		if props.AuthorizationKey != nil {
+		if props.AuthorizationKey != nil && *props.AuthorizationKey != "*****************" {
```

Comment:
Why don't we ignore those invalid value to be specified at the first place?

Reply:
We cannot ignore it at the beginning. Because "*****************" is returned when authorizationKey is changed from "xxxxx" to nil.

Scenario #1:
when authorizationKey isn't specified in tf, api returns nil

Scenario #2:
When authorizationKey is specified in tf, api returns "xxxxx"

Scenario #3:
When authorizationKey is changed from "xxxx" to nil in tf, api returns "*****************"